### PR TITLE
Fix/ Deselects a piece when clicking on another

### PR DESCRIPTION
**Pull request by nischal-shetty2**
Fixes https://github.com/code100x/chess/issues/99

currently you have to click on the same piece to deselect it

solution:
click on any piece and previous one gets deselected

https://github.com/code100x/chess/assets/136328410/4b78d791-5594-4617-808f-62502ebd8dde

### CODE REVIEW

File: apps/frontend/src/components/ChessBoard.tsx [hunk 1]
```diff
@@ -186,8 +186,17 @@ export const ChessBoard = ({
                         }
                         if (!from && square?.color !== chess.turn()) return;
                         if (!isMyTurn) return;
+
                         if (from === squareRepresentation) {
                           setFrom(null);
+                        } else {
+                          setFrom(squareRepresentation);
+
+                          setLegalMoves(
```
Comment (hkirat) [inline apps/frontend/src/components/ChessBoard.tsx:195]: Could u confirm this works if u move ur peice?
It seems like we override from even when the user might be moving somewhere where their piece isnt present

Comment (nischal-shetty2) [reply apps/frontend/src/components/ChessBoard.tsx]: https://github.com/code100x/chess/assets/136328410/b4420e28-0e5a-47ae-8177-1db0e1f1cdf9

you can move the piece i even added a feature where if you click anywhere else on the board it deselects 
there was a small bug of the board displaying invalid moves but i fixed that as well you can review it

**Comment by hkirat** [comment]
merging and trying locally

**Comment by hkirat** [comment]
Looks right 
/bounty $10

**Pull request merged**